src/batch/src/executor/update.rs

```diff
@@ -302,42 +333,40 @@ mod tests {
             );
         });
 
+        assert_matches!(reader.next().await.unwrap()?, TxnMsg::Begin(_));
+
         // Read
         // As we set the chunk size to 5, we'll get 2 chunks. Note that the update records for one
         // row cannot be cut into two chunks, so the first chunk will actually have 6 rows.
         for updated_rows in [1..=3, 4..=5] {
-            let chunk = reader.next().await.unwrap()?;
-
-            assert_eq!(
-                chunk.chunk.ops().chunks(2).collect_vec(),
-                vec![&[Op::UpdateDelete, Op::UpdateInsert]; updated_rows.clone().count()]
-            );
-
-            assert_eq!(
-                chunk.chunk.columns()[0]
-                    .as_int32()
-                    .iter()
-                    .collect::<Vec<_>>(),
-                updated_rows
-                    .clone()
-                    .flat_map(|i| [i * 2 - 1, i * 2]) // -1, +2, -3, +4, ...
-                    .map(Some)
-                    .collect_vec()
-            );
-
-            assert_eq!(
-                chunk.chunk.columns()[1]
-                    .as_int32()
-                    .iter()
-                    .collect::<Vec<_>>(),
-                updated_rows
-                    .clone()
-                    .flat_map(|i| [i * 2, i * 2 - 1]) // -2, +1, -4, +3, ...
-                    .map(Some)
-                    .collect_vec()
-            );
+            assert_matches!(reader.next().await.unwrap()?, TxnMsg::Data(_, chunk) => {
```

Comment:
Perhaps we can derive `EnumAsInner` for `TxnMsg` and call `into_data` here to simplify the tests.